src/shippingState.js

```diff
@@ -67,53 +68,70 @@ export const updateShippingStateAddBagToBox = (boxId, bagId, boxToUpdate) => {
  * Build the bag and specimen data from detailedProviderBoxes data: availableCollectionsObj[bagId]: ['tube1Id', 'tube2Id', 'tube3Id']
  * Isolate the last four digits of tubeId to match availableCollectionsObj format.
  * Move object into availableCollectionsObj
+ * If object is 'unlabelled', it is a stray and it belongs in the 'unlabelled' section of availableCollectionsObj.
  * @param {*} boxId - the box with the bag being removed 
  * @param {*} bagsToMove - the bags being removed from the box
  */
-const addBagToAvailableCollections = (boxId, bagsToMove) => {
+const addBagToAvailableCollections = (boxId, bagId, bagsToMove) => {
     const availableCollectionsObj = appState.getState().availableCollectionsObj;
     const detailedProviderBoxes = appState.getState().detailedProviderBoxes;
 
-    for (const bagId of bagsToMove) {
-        const collectionToMove = detailedProviderBoxes[boxId][bagId].arrElements;
-        const tubeIdArray = collectionToMove.map(tubeId => tubeId.slice(-4));
-
-        availableCollectionsObj[bagId] = tubeIdArray;
+    if (bagId === 'unlabelled') {
+        const collectionToMove = detailedProviderBoxes[boxId]['unlabelled'].arrElements;
+        availableCollectionsObj['unlabelled'].push(...collectionToMove);
+    } else {
+        for (const bagLabel of bagsToMove) {
+            const collectionToMove = detailedProviderBoxes[boxId][bagLabel].arrElements;
+            const tubeIdArray = collectionToMove.map(tubeId => tubeId.slice(-4));
+    
+            availableCollectionsObj[bagLabel] = tubeIdArray;
+        }
     }
-
+    
     appState.setState({
         availableCollectionsObj: {...availableCollectionsObj},
     });
 }
 
 // Remove the bag from the box when user has clicked 'remove bag' in the 'View Shipping Box Contents' table.
-const removeBagFromBox = (boxId, bagsToMove) => {
+const removeBagFromBox = (boxId, bagId, bagsToMove) => {
     const allBoxesList = appState.getState().allBoxesList;
     const boxIndex = allBoxesList.findIndex(box => box[conceptIds.shippingBoxId] === boxId);
 
     if (boxIndex !== -1) { 
-        for (const bagId of bagsToMove) {
-            delete allBoxesList[boxIndex].bags[bagId];
+        if (bagId === 'unlabelled') {
+            bagsToMove = ['unlabelled'];
         }
 
+        bagsToMove.forEach(bagLabel => {
+            delete allBoxesList[boxIndex].bags[bagLabel];
+        });
+
         appState.setState({
             allBoxesList: [...allBoxesList],
         });
     }
 }
 
 // Remove the bag from the availableCollectionsObj when it has been added to a box.
+// Check if bagId is a direct key in availableCollectionsObj. If not, check if it's and element in the 'unlabelled' array.
 const removeBagFromAvailableCollections = (bagId) => {
     const availableCollectionsObj = appState.getState().availableCollectionsObj;
-    delete availableCollectionsObj[bagId];
+
+    if (availableCollectionsObj.hasOwnProperty(bagId)) {
```

Comment:
suggest using `in` instead of `hasOwnProperty`. 
Ex:
```
if (bagId in availableCollectionsObj) {
    delete availableCollectionsObj[bagId];
}
```

Reply:
Updated and tested.